### ISSUE TITLE
Mono.CSharp.AsNonAssignStatementExpression.DoFlowAnalysis  = System.NotImplementedException

### ISSUE THREAD

**Issue by sushihangover** [PlayScriptRedux/playscript#58]
away3d-core-fp11 regression due to the newer FlowAnalysis in mcs:

Mono.CSharp.InternalErrorException:  Mono.CSharp.AsNonAssignStatementExpression.DoFlowAnalysis 
System.NotImplementedException: The requested feature is not implemented.

```
Unhandled Exception:
Mono.CSharp.InternalErrorException: /Users/administrator/Documents/Code/playscript/playscriptredux/away3d-core-fp11/src/away3d/animators/SkeletonAnimator.as(141,19): away3d.animators.SkeletonAnimator.play(string, away3d.animators.transitions.IAnimationTransition, double) ---> Mono.CSharp.InternalErrorException: /Users/administrator/Documents/Code/playscript/playscriptredux/away3d-core-fp11/src/away3d/animators/SkeletonAnimator.as(142,3): ---> System.NotImplementedException: The requested feature is not implemented.
  at Mono.CSharp.AsNonAssignStatementExpression.DoFlowAnalysis (Mono.CSharp.FlowAnalysisContext fc) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Statement.FlowAnalysis (Mono.CSharp.FlowAnalysisContext fc) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Block.DoFlowAnalysis (Mono.CSharp.FlowAnalysisContext fc, Int32 startIndex) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Block.DoFlowAnalysis (Mono.CSharp.FlowAnalysisContext fc) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Statement.FlowAnalysis (Mono.CSharp.FlowAnalysisContext fc) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.If.DoFlowAnalysis (Mono.CSharp.FlowAnalysisContext fc) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Statement.FlowAnalysis (Mono.CSharp.FlowAnalysisContext fc) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Block.DoFlowAnalysis (Mono.CSharp.FlowAnalysisContext fc, Int32 startIndex) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Block.DoFlowAnalysis (Mono.CSharp.FlowAnalysisContext fc) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.ParametersBlock.DoFlowAnalysis (Mono.CSharp.FlowAnalysisContext fc) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Statement.FlowAnalysis (Mono.CSharp.FlowAnalysisContext fc) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.ToplevelBlock.Resolve (Mono.CSharp.BlockContext bc, IMethodData md) [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at Mono.CSharp.ToplevelBlock.Resolve (Mono.CSharp.BlockContext bc, IMethodData md) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.MethodData.Emit (Mono.CSharp.TypeDefinition parent) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.MethodOrOperator.Emit () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Method.Emit () [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at Mono.CSharp.Method.Emit () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.TypeDefinition.Emit () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.ClassOrStruct.Emit () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Class.Emit () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.TypeDefinition.EmitContainer () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.TypeContainer.EmitContainer () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.NamespaceContainer.EmitContainer () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.TypeContainer.EmitContainer () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.NamespaceContainer.EmitContainer () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.TypeContainer.EmitContainer () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.ModuleContainer.EmitContainer () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.AssemblyDefinition.Emit () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.AssemblyDefinitionStatic.Emit () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Driver.Compile () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Driver.Main (System.String[] args) [0x00000] in <filename unknown>:0 
```

**Comment by sushihangover** [comment]
#### New test for testing future regression on this:

as/test-as-AsNonAssignStatementExpression.as

```
package {

    public class Flow {
        public static function Main():int {
                var o:A = new A();

                // Does not become a AsNonAssignStatementExpression
                var i:int = o.B;
                if (i != 99) return 1;
                var x = o.B;
                if (x != 99) return 2;

                // Legal in ActionScript
                // Becomes AsNonAssignStatementExpression
                o.B;

                if (o.refCount != 3) return 99;
                return 0;
        }
    }

}

class A {
        public var refCount:int = 0;
        public function get B():int {
                trace("inside A.B getter");
                refCount++;
                return 99;
        }
}
```

**Comment by sushihangover** [comment]
commit 0b7bff8f4234e3ff762b46d7a8df00b7c1d767f7
Author: SushiHangover sushihangover@outlook.com
Date:   Fri Aug 14 20:01:03 2015 -0700

```
Fixed FlowAnalysis on AsNonAssignStatementExpressions
Closes #58
```
